ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -98,53 +97,89 @@ export default Route.extend(UnloadModelRoute, {
     if (modelType === 'pki-certificate') {
       secret = secret.replace('cert/', '');
     }
-    return hash({
-      secret: this.store
-        .queryRecord(modelType, { id: secret, backend })
-        .then(secretModel => {
-          if (modelType === 'secret-v2') {
-            let targetVersion = parseInt(params.version || secretModel.currentVersion, 10);
-            let version = secretModel.versions.findBy('version', targetVersion);
-            // 404 if there's no version
-            if (!version) {
-              let error = new DS.AdapterError();
-              set(error, 'httpStatus', 404);
-              throw error;
-            }
-            secretModel.set('engine', backendModel);
-
-            return version.reload().then(() => {
-              secretModel.set('selectedVersion', version);
-              return secretModel;
-            });
-          }
-          return secretModel;
-        })
-        .catch(err => {
-          //don't have access to the metadata, so we'll make
-          //a stub metadata model and try to load the version
-          if (modelType === 'secret-v2' && err.httpStatus === 403) {
-            let secretModel = this.store.createRecord('secret-v2');
-            secretModel.setProperties({
-              engine: backendModel,
-              id: secret,
-              // so we know it's a stub model and won't be saving it
-              // because we don't have access to that endpoint
-              isStub: true,
-            });
-            let targetVersion = params.version ? parseInt(params.version, 10) : null;
-            let versionId = targetVersion ? [backend, secret, targetVersion] : [backend, secret];
-            return this.store
-              .findRecord('secret-v2-version', JSON.stringify(versionId), { reload: true })
-              .then(versionModel => {
-                secretModel.set('selectedVersion', versionModel);
-                return secretModel;
-              });
-          }
+    let secretModel;
+
+    let capabilities = this.capabilities(secret);
+    try {
+      secretModel = await this.store.queryRecord(modelType, { id: secret, backend });
+    } catch (err) {
+      if (err.httpStatus === 403 && (modelType === 'secret-v2' || modelType === 'secret')) {
+        await capabilities;
+        // can't read the path and don't have update permission, so re-throw
+        if (!capabilities.get('canUpdate') && modelType === 'secret') {
           throw err;
-        }),
-      capabilities: this.capabilities(secret),
-    });
+        }
+        // don't have access to the metadata for v2 or the secret for v1,
+        // so we make a stub model
+        secretModel = this.store.createRecord(modelType);
+        secretModel.setProperties({
+          id: secret,
+          // so we know it's a stub model and won't be saving it
+          // because we don't have access to that endpoint
+          isStub: true,
```

Comment:
is there a way to make this more clear? `isStub` makes me think fake or incomplete data but doesn't make my mind go all the way to "doesn't have permission to see this"

Reply:
Do you have suggestions about how we might change it? It is fake / incomplete data - we need to return an object so that the template has something to render, so we create a stub model. This code path is only run if you get a 403 from the read endpoint which is what the comment is here: https://github.com/hashicorp/vault/pull/6570/files#diff-66ae76528cc7c3eea9365b29bde71ce9R112

Something like `failedRead` starts to cross over to the `canRead` `canEdit`, etc. properties that the model has so I could see that being confusing too.

Reply:
Do you have suggestions about how we might change it? It is fake / incomplete data - we need to return an object so that the template has something to render, so we create a stub model. This code path is only run if you get a 403 from the read endpoint which is what the comment is here: https://github.com/hashicorp/vault/pull/6570/files#diff-66ae76528cc7c3eea9365b29bde71ce9R112

Something like `failedRead` starts to cross over to the `canRead` `canEdit`, etc. properties that the model has so I could see that being confusing too.